java-frontend/src/test/java/org/sonar/java/model/ExpressionUtilsTest.java

```diff
@@ -284,6 +285,22 @@ void enclosing_method_test() {
     });
   }
 
+  @Test
+  void parent_of_type_method_test() {
+    File file = new File("src/test/files/model/ExpressionEnclosingMethodTest.java");
+    CompilationUnitTree tree = JParserTestUtils.parse(file);
+    FindAssignment findAssignment = new FindAssignment();
+    tree.accept(findAssignment);
+    findAssignment.assignments.forEach(a -> {
+      ClassTree parent = (ClassTree) ExpressionUtils.getParentOfType(a, Tree.Kind.CLASS, Tree.Kind.INTERFACE);
+      if (parent.is(Tree.Kind.CLASS)) {
+        assertThat(parent.simpleName().name()).isEqualTo("Clazz");
+      } else if (parent.is(Tree.Kind.INTERFACE)) {
+        assertThat(parent.simpleName().name()).isEqualTo("I1");
+      }
+    });
```

Comment:
Because the method `getParentOfType` is added to the frontend module API, we should cover all branches here.
The current test only validates use cases that are related to S1656 but we should also test other usages of the API.
```suggestion
    });
   assertThat(ExpressionUtils.getParentOfType(tree, Tree.Kind.COMPILATION_UNIT)).isNull();
```